block_device.go

```diff
@@ -45,6 +52,14 @@ func (bd *BlockDevice) Write(p []byte) (int, error) {
 			return 0, err
 		}
 
+		// This will only succeed on UBI volumes, fail silently on SD/MMC
+		//
+		// This must be here becuase UBI will verify if data written
+		// matches the size argument when the file descriptor is closed.
+		syscall.Syscall(syscall.SYS_IOCTL, out.Fd(),
+			uintptr(unsafe.Pointer(UBI_IOCVOLUP)),
+			uintptr(unsafe.Pointer(&bd.ImageSize)))
```

Comment:
That makes lot of sense. I was thinking more about adding some run time check here. Maybe something similar to `if strings.HasPrefix(inactivePartition, "/dev/") == false`?